go/arrow/array/table.go

```diff
@@ -92,6 +92,8 @@ type simpleTable struct {
 	schema *arrow.Schema
 }
 
+var _ arrow.Table = (*simpleTable)(nil)
```

Comment:
This is a compile-time check of `simpleTable` against the `arrow.Table` interface.